File: SYP.Web/Modules/ServerTypes/Setting/VatRatesEditorForm.ts
import { BooleanEditor, initFormType, IntegerEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface VatRatesEditorForm {
    Name: StringEditor;
    Code: StringEditor;
    IsDefault: BooleanEditor;
    IsActive: BooleanEditor;
    SortOrder: IntegerEditor;
}

export class VatRatesEditorForm extends PrefixedContext {
    static readonly formKey = 'Setting.VatRatesEditor';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!VatRatesEditorForm.init) {
            VatRatesEditorForm.init = true;

            var w0 = StringEditor;
            var w1 = BooleanEditor;
            var w2 = IntegerEditor;

            initFormType(VatRatesEditorForm, [
                'Name', w0,
                'Code', w0,
                'IsDefault', w1,
                'IsActive', w1,
                'SortOrder', w2
            ]);
        }
    }
}

File: SYP.Web/Modules/Setting/VatRates/Editor/VatRatesEditorForm.cs
namespace SYP.Setting.Forms;

[FormScript("Setting.VatRatesEditor")]
[BasedOnRow(typeof(VatRatesRow), CheckNames = true)]
public class VatRatesEditorForm
{
    public string Name { get; set; }
    public string Code { get; set; }
    public bool IsDefault { get; set; }
    public bool IsActive { get; set; }
    public int SortOrder { get; set; }
}